src/Microsoft.Health.Fhir.Shared.Web/Startup.cs

```diff
@@ -111,6 +116,19 @@ private void AddTaskHostingService(IServiceCollection services)
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
         public virtual void Configure(IApplicationBuilder app)
         {
+            app.Use(async (context, next) =>
+            {
+                if (instanceId != null)
+                {
+                    string instanceKey = "X-Instance-Id";
```

Comment:
Perhaps add a new constant to the class KnownHeaders instead of hard coding the string.

Reply:
done